### ISSUE TITLE
Aborting a manual sync does abort but dialog stays

### ISSUE THREAD

**Issue by cffm** [Razzeee/script.trakt#41]
When you do a manual sync and want to abort it (click the cancel button), it will abort as far as the log file says. The dialog that shows the status stays (status whatever it was with "....canceling"). I waited for 10 minutes, log file reported that it canceled and was done.

**Comment by N3MIS15** [comment]
If there are no movies in your library, disable the sync movies settings. That is a known crash that needs fixing.

**Comment by dpons039** [comment]
Hi,

This happens if there's not  TV Shows as well.

**Comment by DejaVu** [comment]
Got over 800 Movies and 3,400 TV Shows in my library.

JSON query works - and is massive - but sending the information fails.

Can paste the full log if it'll be more helpful?

**Comment by nate1280** [comment]
Looking at that log file, I don't even know what plugin you have, because those line numbers don't even match up to this plugin's code base.

```
self.xbmc_movies = xbmc_json({"jsonrpc": "2.0", "method": "VideoLibrary.GetMovies", "params": {"properties": ["title", "imdbnumber", "year", "playcount"]}, "id": 0})['movies']
```

That piece of code doesn't exist like that in this plugin.

Also, it seems like you're using a different plugin all together, your folder name is "script.trakt.sync-master", which seems to be the sync code in plugin form before it was merged with this plugin.

I would recommend deleting that plugin, and re-downloading this one, and trying again.

**Comment by DejaVu** [comment]
It's the Zip file from the master on rectifyer's git. Downloaded and installed as Zip.

https://github.com/rectifyer/script.trakt/archive/master.zip

**Comment by nate1280** [comment]
This part here:

```
02:52:17 T:2856319808   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: ('movies',)
                                            Traceback (most recent call last):
                                              File "/home/xbmc/.xbmc/addons/script.trakt.sync-master/default.py", line 594, in <module>
                                                movies.Run()
                                              File "/home/xbmc/.xbmc/addons/script.trakt.sync-master/default.py", line 301, in Run
                                                self.GetFromXBMC()
                                              File "/home/xbmc/.xbmc/addons/script.trakt.sync-master/default.py", line 157, in GetFromXBMC
                                                self.xbmc_movies = xbmc_json({"jsonrpc": "2.0", "method": "VideoLibrary.GetMovies", "params": {"properties": ["title", "imdbnumber", "year", "playcount"]}, "id": 0})['movies']
                                            KeyError: ('movies',)
                                            -->End of Python script error report<--
```

To me, the addon folder is "script.trakt.sync-master", and the error occurs in the file "default.py"

I downloaded a zip from this repo, and the folder name it would create is "script.trakt-master" and has no "default.py" file in it.  So the above error is coming from another plugin.

So, unless there is another error in the log file, you may want to check your plugins to make sure you don't have any other trakt type plugins running and interfering.

**Comment by N3MIS15** [comment]
Script.trakt.sync is the old addon i made before reworking the code and merging it in justins script.trakt

**Comment by nate1280** [comment]
Might want to re-download the latest master, and try again, lots of changes have been made since you posted this.

**Comment by DejaVu** [comment]
I think I worked out what my problem is, but has yet to have been reported. 
I was using a local library fine and it was doing it's thing perfectly. 

As soon as I transfered my library back to MySQL (*without Movies in it) it broke. 
I think it's either something to do with MySQL or the already submitted problem when not having both TV Shows AND Movies in the library.

**Comment by nate1280** [comment]
The issue when not having tv shows and/or movies has been sorted, and I use MySQL as my backend with no issues.

If you have more logs, post em.

**Comment by rectifyer** [comment]
Seems to be resolved.

**Comment by Razzeee** [comment]
@cffm 
the handling of a cancled sync is very weird for me, thats probably something we can do, when we have pushed the first version to the people waiting for it.

**Comment by cffm** [comment]
‎Wow. The man of the trakt messiah group! Thank you for answering :)It canceled fine. But the dialog just kept blocking the screen. So maybe after you add the line "canceled manual sync", remove the dialog. But this is rare. Was caused by a weird bug where it would do a manual synch after I watched something via salts. It scrobbled but instead of marking ‎it read, it initiated a full synch. After the 5th full synch I aborted. And that did not work :)Thank you! Awesome work! Like a zillion check github every hour to be cutting edge (vs nothing :)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   From: RazzeeeSent: Wednesday, January 28, 2015 11:19 PMTo: Razzeee/script.traktReply To: Razzeee/script.traktCc: cffmSubject: Re: [script.trakt] Aborting a manual sync does abort but dialog stays (#41)@cffm 
the handling of a cancled sync is very weird for me, thats probably something we can do, when we have pushed the first version to the people waiting for it.

—Reply to this email directly or view it on GitHub.
